src/v2/models/AppState.js

```diff
@@ -238,38 +242,43 @@ export default Model.extend({
 
   clearAppStateCache() {
     // clear appState before setting new values
-    this.clear({ silent: true });
+    const attrs = {};
+    for (var key in this.attributes) {
+      if (key !== 'currentFormName') {
+        attrs[key] = void 0;
+      }
+    }
+    this.set(attrs, Object.assign({}, { unset: true, silent: true }));
```

Comment:
since the error handling part of the widget is now changed and it accepts the errors as idx responses.. For certain flows we don't want `currentFormName` to get cleared.. Because in error states we would need this formName so that view can get render as expected.. Also for `handlSwitchForm` as far as i remember

Reply:
also form name change can cause unexpected `re-render`